blaze/compute/mongo.py

```diff
@@ -57,27 +58,33 @@
 import toolz
 import datetime
 
-from ..expr import (var, Label, std, Sort, count, nunique, nelements, Selection,
+from ..expr import (Sort, count, nunique, nelements, Selection,
                     mean, Reduction, Head, ReLabel, Distinct, ElemWise, By,
                     Symbol, Projection, Field, sum, min, max, Gt, Lt, Ge, Le,
                     Eq, Ne, And, Or, Summary, Like, Broadcast, DateTime,
                     Microsecond, Date, Time, Expr, symbol, Arithmetic, floor,
                     ceil, FloorDiv)
+from ..expr.broadcast import broadcast_collect
 from ..expr import math
 from ..expr.datetime import Day, Month, Year, Minute, Second, UTCFromTimestamp
 from ..compatibility import _strtypes
-from .core import compute
 
 from ..dispatch import dispatch
 
 
 __all__ = ['MongoQuery']
 
+
 @dispatch(Expr, Collection)
 def pre_compute(expr, data, scope=None, **kwargs):
     return MongoQuery(data, [])
 
 
+@dispatch(Expr, Database)
+def pre_compute(expr, data, **kwargs):
+    return data
```

Comment:
Yes because of the annoying `__iter__` definition on `Database` objects. The pure python backend has a `pre_compute` implementation that dispatches on `Sequence`.